eventlet/green/thread.py

```diff
@@ -32,9 +32,12 @@ def _count():
 
 
 def get_ident(gr=None):
-    if gr is None:
-        return id(greenlet.getcurrent())
-    else:
+    try:
+        if gr is None:
+            return id(greenlet.getcurrent())
+        else:
+            return id(gr)
+    except:
```

Comment:
On the other hand I wonder if by forcing the return an identifier even when `gr` is equal to `None` we could may find us hidding these symptoms. Symptoms which may be the manifestation of a more pernicious problem.

Please can you give us more details about the context where you face this problem?
Do you have a reproducer?

Reply:
That's more or less the scenario you propose to follow.

Reply:
As this eventlet module (`eventlet.green.thread`) is the patched version of the Python thread module I'd argue that maybe, one option, could be to return the result of the native `get_ident` function. It will return the ‘thread identifier’ of the current thread. IMO, That would more meaningful that returning an integer computed from a None value.

https://docs.python.org/3/library/threading.html#threading.get_ident

But I still continue to think that the observed behavior is the symptom of a problem more deeply rooted.

Reply:
thank you for your review, after searching for root of this problem, I found its not related to this package, and its about bobcat keystone package that uses this package. and I tried to fix actual problem and eventlet package works fine.

for reproduce this package install openstack prereqs and set bobcat, then install keystone, and run su -s /bin/sh -c "keystone-manage db_sync", in ubuntu 22.04, you can see error.
thank you again 🙌

Reply:
Thanks for your feedback. So, if you agree with abandoning this proposal I think we can close this github pull request and its associated github issue. Do you agree?